### PR TITLE
[lldb-dap] Refactor lldb-dap.cpp to not use global DAP variable.

### PR DESCRIPTION
**Pull request by ashgti**
This removes the global DAP variable and instead allocates a DAP instance in main. This should allow us to refactor lldb-dap to enable a server mode that accepts multiple connections.

### CODE REVIEW

**Review by JDevlieghere — APPROVED** [review]
🥳

File: lldb/tools/lldb-dap/lldb-dap.cpp [hunk 1]
```diff
@@ -5046,30 +5026,50 @@ int main(int argc, char *argv[]) {
...
+  // Initialize LLDB first before we do anything.
+  lldb::SBDebugger::Initialize();
+
+  // Terminate the debugger before the C++ destructor chain kicks in.
+  auto terminate_debugger =
+      llvm::make_scope_exit([] { lldb::SBDebugger::Terminate(); });
+
+  DAP dap;
+  dap.debug_adaptor_path = program_path.str().str();
+  dap.repl_mode = defaultReplMode;
```
Comment (walter-erquinigo) [inline lldb/tools/lldb-dap/lldb-dap.cpp:5039]: can you add the debug_adaptor_path and the repl_mode in the constrsuctor of the Dap class?

**Review by walter-erquinigo — APPROVED** [review]

File: lldb/tools/lldb-dap/lldb-dap.cpp (hunk 1 above)
Comment (ashgti) [reply lldb/tools/lldb-dap/lldb-dap.cpp]: Done, moved those both to the constructor.

File: lldb/tools/lldb-dap/lldb-dap.cpp [hunk 2]
```diff
@@ -5046,30 +5026,48 @@ int main(int argc, char *argv[]) {
...
 #endif
+
+  // Initialize LLDB first before we do anything.
+  lldb::SBDebugger::Initialize();
+
+  // Terminate the debugger before the C++ destructor chain kicks in.
+  auto terminate_debugger =
+      llvm::make_scope_exit([] { lldb::SBDebugger::Terminate(); });
+
+  DAP dap = DAP(program_path.str(), default_repl_mode);
```
Comment (labath) [inline lldb/tools/lldb-dap/lldb-dap.cpp:5037]: Should the dap object be copyable? Given that things hold references to it, maybe the copy operations should be deleted (and this object constructed in place) ?

Comment (ashgti) [reply lldb/tools/lldb-dap/lldb-dap.cpp:5037]: I can follow up with a delete of the copy constructor

**Pull request merged**